src/planner/World.cpp

```diff
@@ -171,10 +171,27 @@ bool World::State::operator!=(const State& other) const
 //==============================================================================
 World::State World::getState() const
 {
-  World::State state;
+  std::vector<std::string> names;
+  names.reserve(getNumSkeletons());
 
   for (const auto& skeleton : mSkeletons)
-    state.configurations[skeleton->getName()] = skeleton->getConfiguration();
+    names.push_back(skeleton->getName());
+
+  return getState(names);
+}
+
+//==============================================================================
+World::State World::getState(std::vector<std::string>& names) const
+{
+  using ConfigFlags = dart::dynamics::Skeleton::ConfigFlags;
+
+  World::State state;
+
+  for (const auto& name : names)
+  {
+    state.configurations[name]
+        = getSkeleton(name)->getConfiguration(ConfigFlags::CONFIG_POSITIONS);
```

Comment:
This can segfault when the world doesn't contain `name` skeleton. It'd be safer to insert the configuration only when `getSkeleton(name)` is not `nullptr`.